flow/record/adapter/sqlite.py

```diff
@@ -0,0 +1,246 @@
+from __future__ import annotations
+
+import logging
+import sqlite3
+from datetime import datetime
+from typing import Iterator
+
+from flow.record import Record, RecordDescriptor
+from flow.record.adapter import AbstractReader, AbstractWriter
+from flow.record.base import RESERVED_FIELDS
+
+logger = logging.getLogger(__name__)
+
+__usage__ = """
+SQLite adapter
+---
+Write usage: rdump -w sqlite://[PATH]
+Read usage: rdump sqlite://[PATH]
+[PATH]: path to sqlite database file.
+"""
+
+# flow.record field mappings to SQLite types
+FIELD_MAP = {
+    "int": "INTEGER",
+    "uint32": "INTEGER",
+    "varint": "INTEGER",
+    "float": "REAL",
+    "boolean": "INTEGER",
+    "bytes": "BLOB",
+    "filesize": "INTEGER",
+    "datetime": "TIMESTAMP",
+}
+
+
+# SQLite types to flow.record field mappings
+SQLITE_FIELD_MAP = {
+    "VARCHAR": "string",
+    "INTEGER": "varint",
+    "BLOB": "bytes",
+    "REAL": "float",
+    "DOUBLE": "float",
+    "BOOLEAN": "boolean",
+    "DATETIME": "datetime",
+    "TIMESTAMP": "datetime",
+}
+
+
+def sanitized_name(name: str) -> str:
+    """Returns a sanitized version of name.
+
+    Some (field) names are not allowed in flow.record, while they can be allowed in SQLite.
+    This sanitizes the name so it can still be used in flow.record.
+
+    >>> sanitized_name("my-variable-name-with-dashes")
+    'my_variable_name_with_dashes'
+    >>> sanitized_name("_my_name_starting_with_underscore")
+    'n__my_name_starting_with_underscore'
+    >>> sanitized_name("1337")
+    'n_1337'
```

Comment:
```suggestion
        >>> sanitized_name("my-variable-name-with-dashes")
        'my_variable_name_with_dashes'
        >>> sanitized_name("_my_name_starting_with_underscore")
        'n__my_name_starting_with_underscore'
        >>> sanitized_name("1337")
        'n_1337'
```
Might be good to check but I believe this way it also gets formatted nicely in Sphinx.